tests/android-test.ts

```diff
@@ -0,0 +1,146 @@
+    diffString += "<string name=\"login_epilogue_screen_title\">Logged in as</string>\n";
+    diffString += customDiffs;
+
+    return diffString;
+}
+
+beforeEach(() => {
+    dm.warn = jest.fn().mockReturnValue(true);
+});
+
+describe("string checks", () => {
+    it("Warn when an entry which references another string resources is added without the translatable=false attribute", async () => {
+        // Create mocks
+        const mockDiffFromFile = await getMockedDiff("+++ <string name=\"login_prologue_screen_title\">@string/app_name</string>");
+        dm.danger = {
+            github: { pr: { base: { ref: "a-branch" } } },
+            git: {
+                modified_files: ["values/strings.xml"],
+                diffForFile: async () => { return { added: mockDiffFromFile } }
+            }
+        };
+        
+        // Run the checks
+        await androidChecks();
+        
+        // Check that the instructions appear correct.
+        expect(dm.warn).toHaveBeenCalledWith(expect.stringContaining("This PR adds a translatable entry to \`strings.xml\` which references another string resource: this usually causes issues with translations."));
+    })
+
+    it("Don't warn when an entry which references another string resources is added with the translatable=false attribute", async () => {
+        // Create mocks
+        const mockDiffFromFile = await getMockedDiff("+++ <string name=\"login_prologue_screen_title\" translatable=\"false\">@string/app_name</string>");
+        dm.danger = {
+            github: { pr: { base: { ref: "a-branch" } } },
+            git: {
+                modified_files: ["values/strings.xml"],
+                diffForFile: async () => { return { added: mockDiffFromFile } }
+            }
+        };
+        
+        // Run the checks
+        await androidChecks();
+        
+        // Check that the no message is shown.
+        expect(dm.warn).not.toHaveBeenCalledWith();
+    })
+
+    it("Don't warn when strings resources are added to strings.xml", async () => {
+        // Create mocks
+        const mockDiffFromFile = await getMockedDiff("+++ <string name=\"login_prologue_screen_title\">A test string</string>");
+        dm.danger = {
+            github: { pr: { base: { ref: "a-branch" } } },
+            git: {
+                modified_files: ["values/strings.xml"],
+                diffForFile: async () => { return { added: mockDiffFromFile } }
+            }
+        };
+        
+        // Run the checks
+        await androidChecks();
+        
+        // Check that the no message is shown.
+        expect(dm.warn).not.toHaveBeenCalledWith();
+    })
+
+    it("Don't warn when there are no changes in strings.xml", async () => {
+        // Create mocks
+        dm.danger = {
+            github: { pr: { base: { ref: "a-branch" } } },
+            git: {
+                modified_files: ["a-file.kt"],
+                diffForFile: async () => { return { added: '' } }
```

Comment:
Maybe provide a dummy diff for the mock here, so that it at least simulates an actual change in the `.kt` file?

Otherwise that test could go green if our implementation were to erroneously test all the files (and not just `strings.xml`) but skip files with empty diffs, seeing them as unchanged, while the real reason why we want it to be green is because there's no `strings.xml`, not that there are not changes in any files including kt files.

```suggestion
                diffForFile: async () => { return { added: '+++ getString(R.string.example),' } }
```